Sharpmake/Project.Configuration.cs

```diff
@@ -3244,16 +3246,15 @@ internal void Link(Builder builder)
                                     dependencySetting.HasFlag(DependencySetting.ForceUsingAssembly))
                                     AdditionalUsingDirectories.Add(dependency.TargetPath);
 
-                                bool? referenceOutputAssembly = ReferenceOutputAssembly;
+                                bool? referenceOutputAssembly = dependency.ReferenceOutputAssembly;
```

Comment:
Hum, do we really want to look for the ReferenceOutputAssembly property on the dependency? Or does it make more sense on the current project itself?